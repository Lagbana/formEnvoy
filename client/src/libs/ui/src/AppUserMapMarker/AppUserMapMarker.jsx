/* User's map marker
=============================== */

import React from 'react';
import { StyleSheet, View, Animated, Easing, ViewStyle } from 'react-native';
import { Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
export interface AppUserMapMarkerProps {
	color?: string;
	style?: ViewStyle;
}

export const AppUserMapMarker: React.FC<AppUserMapMarkerProps> = ({ color = 'blue', ...props }) => {
  // vars

  const animValue = new Animated.Value(0);
  const DOT_SIZE = 28;
  const RING_WIDTH = 2;

  // animations

  Animated.loop(
    Animated.sequence([
      Animated.delay(500),
      Animated.timing(animValue, {
        toValue: 1,
        duration: 1000,
        useNativeDriver: false,
        easing: Easing.linear,
      }),
    ])
  ).start();

  // styles

  const styles = StyleSheet.create({
    container: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 5 },
      shadowRadius: 5,
      shadowOpacity: 0.25,
      padding: 15,
      elevation: 4,
    },
    dot: {
      width: DOT_SIZE,
      height: DOT_SIZE,
      backgroundColor: Colors[color],
      borderRadius: DOT_SIZE,
      zIndex: 1,
    },
    ring: {
      width: DOT_SIZE,
      height: DOT_SIZE,
      position: 'absolute',
      top: (DOT_SIZE + RING_WIDTH) / 2,
      left: (DOT_SIZE + RING_WIDTH) / 2,
      borderRadius: DOT_SIZE,
      borderWidth: RING_WIDTH,
      borderColor: Colors[color],
      zIndex: 0,
    },
  });

  const scale = animValue.interpolate({
    inputRange: [0, 1],
    outputRange: [1, 2],
  });

  const opacity = animValue.interpolate({
    inputRange: [0, 0.5, 1],
    outputRange: [0, 0.5, 0],
  });

  return (
    <View style={[styles.container, props.style]}>
      <View style={styles.dot} />
      <Animated.View
        style={[styles.ring, { opacity, transform: [{ scale }] }]}
      />
    </View>
  );
};
