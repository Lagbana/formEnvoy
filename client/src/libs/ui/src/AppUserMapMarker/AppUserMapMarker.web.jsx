/* User's map marker (no animation)
=============================== */

import React from 'react';
import { StyleSheet, View, ViewStyle } from 'react-native';
import { Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
export interface AppUserMapMarkerProps {
	color: string;
	style: ViewStyle;
}

export const AppUserMapMarker: React.FC<AppUserMapMarkerProps> = ({ color = 'blue', ...props }) => {
  // vars

  const DOT_SIZE = 28;

  // styles

  const styles = StyleSheet.create({
    container: {
      shadowColor: '#000',
      shadowOffset: { width: 0, height: 5 },
      shadowRadius: 5,
      shadowOpacity: 0.25,
      borderRadius: DOT_SIZE,
      elevation: 4,
    },
    dot: {
      width: DOT_SIZE,
      height: DOT_SIZE,
      backgroundColor: Colors[color],
      borderRadius: DOT_SIZE,
      zIndex: 1,
    },
  });

  return (
    <View style={[styles.container, props.style]}>
      <View style={styles.dot} />
    </View>
  );
};
