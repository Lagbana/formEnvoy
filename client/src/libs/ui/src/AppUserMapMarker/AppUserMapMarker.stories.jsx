import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppUserMapMarker } from './AppUserMapMarker'

storiesOf('AppUserMapMarker', module)
  .add('Blue map marker', () => <AppUserMapMarker color={'blue'} />)
  .add('Red map marker', () => <AppUserMapMarker color={'red'} />)
