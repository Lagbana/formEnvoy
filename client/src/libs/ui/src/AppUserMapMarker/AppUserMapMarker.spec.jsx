import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppUserMapMarker } from './AppUserMapMarker'

configure({ adapter: new Adapter() })

test('<AppUserMapMarker /> renders correctly', () => {
	const component = shallow(<AppUserMapMarker color={'blue'} />)
	component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
