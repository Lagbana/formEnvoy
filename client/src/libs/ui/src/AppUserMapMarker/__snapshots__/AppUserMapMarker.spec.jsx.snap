// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppUserMapMarker /> renders correctly 1`] = `""`;
